### PR TITLE
Clean up the Daemonise method a bit.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: common/base/Init.cpp [hunk 1]
```diff
@@ -360,19 +359,27 @@ void Daemonise() {
...
+  int maxfd = sysconf(_SC_OPEN_MAX);
+  if (maxfd == -1) {
+    if (rl.rlim_max == RLIM_INFINITY) {
+      maxfd = 1024;
+    } else {
+      maxfd = rl.rlim_max;
+    }
+  }
+
+  for (unsigned int fd = 0; fd < rl.rlim_max; fd++) {
```
Comment (peternewman) [inline common/base/Init.cpp]: Should this not be maxfd?

**Comment by peternewman** [comment]
Generally LGTM

File: common/base/Init.cpp (hunk 1 above)
Comment (nomis52) [inline common/base/Init.cpp]: Good catch.

Comment (peternewman) [inline common/base/Init.cpp]: I imagine Coverity would have thrown an unused variable warning too.

File: common/base/Init.cpp [hunk 2]
```diff
@@ -360,19 +359,27 @@ void Daemonise() {
...
 
   // close all fds
-  if (rl.rlim_max == RLIM_INFINITY)
-    rl.rlim_max = 1024;
-  for (i = 0; i < rl.rlim_max; i++)
-    close(i);
+  int maxfd = sysconf(_SC_OPEN_MAX);
+  if (maxfd == -1) {
+    if (rl.rlim_max == RLIM_INFINITY) {
+      maxfd = 1024;
```
Comment (peternewman) [inline common/base/Init.cpp]: So why 1024, or is this a 1024 ought to be enough for anyone type situation?

**Comment by peternewman** [comment]
LGTM, merge if my outstanding comment doesn't throw up a new bug.

File: common/base/Init.cpp (hunk 2 above)
Comment (nomis52) [inline common/base/Init.cpp]: Correct. 

For our use case, we only ever have 3 fds open at this point anyway.

**Pull request merged**